### PR TITLE
code cleanup: Split up virtctl Run() commands into separate functions.

### PR DESCRIPTION
**Pull request by prnaraya**
**What this PR does / why we need it**: The Run function in vm.go was quite long, and a few of the commands had already been moved into separate functions. This PR cleans up the rest of the commands, creating separate functions for them as well. Though the cases for `COMMAND_START` and `COMMAND_MIGRATE` are already quite short, they also were put into separate functions to keep things consistent between all the commands. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virtctl/vm/vm.go [hunk 1]
```diff
@@ -511,6 +511,148 @@ func expandVirtualMachine(namespace string, virtClient kubecli.KubevirtClient, o
...
+	err := virtClient.VirtualMachine(namespace).Start(context.Background(), vmiName, &v1.StartOptions{Paused: startPaused, DryRun: *dryRunOption})
+	if err != nil {
+		return fmt.Errorf("Error starting VirtualMachine %v", err)
+	}
+
+	return nil
+}
+
+func stopVirtualMachine(vmiName, namespace string, virtClient kubecli.KubevirtClient, dryRunOption *[]string) error {
+	if gracePeriodIsSet(gracePeriod) && forceRestart == false {
```
Comment (alicefr) [inline pkg/virtctl/vm/vm.go:524]: We have multiple if statements that check `gracePeriodIsSet` can we reduce and simplify them into a single one, maybe avoiding too many nested if statements

File: pkg/virtctl/vm/vm.go [hunk 2]
```diff
@@ -511,6 +511,148 @@ func expandVirtualMachine(namespace string, virtClient kubecli.KubevirtClient, o
...
+	err := virtClient.VirtualMachine(namespace).Stop(context.Background(), vmiName, &v1.StopOptions{DryRun: *dryRunOption})
+	if err != nil {
+		return fmt.Errorf("Error stopping VirtualMachine %v", err)
+	}
+
+	return nil
+}
+
+func restartVirtualMachine(vmiName, namespace string, virtClient kubecli.KubevirtClient, dryRunOption *[]string) error {
+	if gracePeriod != notDefinedGracePeriod && forceRestart == false {
```
Comment (alicefr) [inline pkg/virtctl/vm/vm.go:547]: same as above

File: pkg/virtctl/vm/vm.go (hunk 1 above)
Comment (prnaraya) [reply pkg/virtctl/vm/vm.go]: I've made changes to the stopVM and restartVM functions to simplify the checks of gracePeriodIsSet. Please let me know what you think.

File: pkg/virtctl/vm/vm.go [hunk 3]
```diff
@@ -511,6 +511,152 @@ func expandVirtualMachine(namespace string, virtClient kubecli.KubevirtClient, o
...
+	}
+
+	return nil
+}
+
+func stopVirtualMachine(vmiName, namespace string, virtClient kubecli.KubevirtClient, dryRunOption *[]string) error {
+	if gracePeriodIsSet(gracePeriod) {
+		if !forceRestart {
+			return fmt.Errorf("Can not set gracePeriod without --force=true")
+		} else {
```
Comment (alicefr) [inline pkg/virtctl/vm/vm.go:527]: nit: just avoid this else, you are already returning in the  if statement, we can avoid an additional nested block

**Comment by prnaraya** [comment]
@alicefr 
This PR appears to overlap with #9183 so would you be able to chime in on going about deduplicating these PRs? Thanks

**Comment by 0xFelix** [comment]
As commented on #9183, it would be great if both of you could work on merging the improved functions into the separate command files.

**Comment by prnaraya** [comment]
As discussed in #9183, I will close this PR for now and once that PR is merged, I will open a follow-up PR to clean up the functions a little more.

**Pull request closed without merging**